sdcclient/_monitor.py

```diff
@@ -608,14 +604,18 @@ def create_dashboard_from_template(self, dashboard_name, template, scope, shared
                 'filterNotificationsUserInputFilter': ''
             }
 
-        # set dashboard scope to the specific parameter
-        scopeExpression = self.convert_scope_string_to_expression(scope)
-        if scopeExpression[0] == False:
-            return scopeExpression
-        if scopeExpression[1]:
-            template['scopeExpressionList'] = map(lambda ex: {'operand': ex['operand'], 'operator': ex['operator'], 'value': ex['value'], 'displayName': '', 'variable': False}, scopeExpression[1])
-        else:
-            template['scopeExpressionList'] = None
+        if scope is not None:
+            if isinstance(scope, basestring) == False:
+                return [False, 'Invalid scope format: Expected a string']
+            
+            # set dashboard scope to the specific parameter
+            scopeExpression = self.convert_scope_string_to_expression(scope)
+            if scopeExpression[0] == False:
+                return scopeExpression
+            if scopeExpression[1]:
+                template['scopeExpressionList'] = map(lambda ex: {'operand': ex['operand'], 'operator': ex['operator'], 'value': ex['value'], 'displayName': '', 'variable': False}, scopeExpression[1])
+            else:
+                template['scopeExpressionList'] = None
```

Comment:
I think this should be done in case of `scope == None` as well, otherwise you may end up submitting a template with scope, while the function is expected to override it with the `scope` parameter instead.

Reply:
In my first approach, I called the create_dashboard_from_template method with scope set either as a string (for old format) or a list (for new format). But as the template is created from the dashboard JSON, scopeExpressionList attribute is already set from the imported dashboard. So the only change is not to remove it when scope is None, as inthat case the convert_scope_string_to_expression will return an empty expression, and "None" will be assigned to template['scopeExpressionList', ignoring the scope from the imported dashboard.